src/Common/Error/ErrorFactory.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Sherl\Sdk\Common\Error;
+
+class ErrorFactory
+{
+    private string $domainName;
+    private mixed $errors;
+
+    public function __construct($domainName, $errors = null)
+    {
+        $this->domainName = $domainName;
+        $this->errors = $errors;
+    }
+
+    public function create($code, $data = null): SherlException
```

Comment:
Il faudrait ajouter du typage également dans les arguments des méthodes (incluant le constructeur), par exemple : 

```suggestion
    public function create(string $code, mixed $data = null): SherlException
```